CRM/MembershipExtras/BAO/MembershipPeriod.php

```diff
@@ -21,6 +21,127 @@ public static function create($params) {
     return $instance;
   }
 
+  /**
+   * Creates a new period for the given membership, using the membership's
+   * current state.
+   *
+   * @param int $membershipID
+   *
+   * @return \CRM_MembershipExtras_DAO_MembershipPeriod|NULL
+   */
+  public static function createPeriodForMembership($membershipID) {
+    $membership = civicrm_api3('Membership', 'getsingle', [
+      'id' => $membershipID,
+    ]);
+    $lastActivePeriod = self::getLastActivePeriod($membershipID);
+    $lastPayment = self::getMembershipLastContribution($membershipID);
+
+    return self::create([
+      'membership_id' => $membershipID,
+      'start_date' => self::calculateStartDate($membership, $lastActivePeriod),
+      'end_date' => $membership['end_date'],
+      'payment_entity_table' => self::calculateEntityTableForContribution($lastPayment),
+      'entity_id' => self::calculateEntityIDForContribution($lastPayment),
+      'is_active' => TRUE,
+    ]);
+  }
+
+  /**
+   * Returns an array with the information for the latest active period for the
+   * given membership.
+   *
+   * @param int $membershipID
+   *
+   * @return array
+   */
+  public static function getLastActivePeriod($membershipID) {
+    $membershipPeriod = new self();
+    $membershipPeriod->membership_id = $membershipID;
```

Comment:
I might be mistaken but wouldn't the last active membership has the is_active Flag = TRUE  too ?

Reply:
Yes it would, but that's the idea, to obtain the last period that has is_active = TRUE, even if its the current one.